src/asciidocParser.ts

```diff
@@ -2,26 +2,21 @@ import * as vscode from 'vscode'
 import * as path from 'path'
 import { spawn } from 'child_process'
 
-const asciidoctor = require('@asciidoctor/core')()
+const asciidoctor = require('@asciidoctor/core')
 const docbook = require('@asciidoctor/docbook-converter')
 const kroki = require('asciidoctor-kroki')
 const highlightjsAdapter = require('./highlightjs-adapter')
-highlightjsAdapter.register(asciidoctor)
-
-const useKroki = vscode.workspace.getConfiguration('asciidoc', null).get('use_kroki')
-if (useKroki) {
-  kroki.register(asciidoctor.Extensions)
-}
 
 export class AsciidocParser {
     public html: string = ''
     public document = null
+    public processor = null
     private extPath = vscode.extensions.getExtension('asciidoctor.asciidoctor-vscode').extensionPath
     private stylesdir = path.join(this.extPath, 'media')
 
     constructor (private readonly filename: string, private errorCollection: vscode.DiagnosticCollection = null) {
-      this.filename = filename
-      this.errorCollection = errorCollection
+      this.processor = asciidoctor()
+      highlightjsAdapter.register(this.processor)
```

Comment:
You cannot register the extension more than once. The reason is that we are doing the following in the `register` function:

```
const HighlightjsSyntaxHighlighter = asciidoctor.SyntaxHighlighter.for('highlight.js')
const customHighlightJsAdapter = Opal.klass(Opal.nil, HighlightjsSyntaxHighlighter, 'CustomHighlightJsAdapter')
```

So the first time, we actually get the built-in "HighlightjsSyntaxHighlighter" from Asciidoctor. But, since we replace the built-in syntax highlighter with our "CustomHighlightJsAdapter", the second time the function `asciidoctor.SyntaxHighlighter.for('highlight.js')` will actually return our "CustomHighlightJsAdapter".

That's the reason why I'm doing the following in #451:

```js
// top definition, called only once
const highlightjsBuiltInSyntaxHighlighter = asciidoctor.SyntaxHighlighter.for('highlight.js')

// ...

// can be called multiple times
highlightjsAdapter.register(highlightjsBuiltInSyntaxHighlighter, context, editor)
```


**src/highlightjs-adapter.ts**

```js
module.exports.register = (highlightjsBuiltInSyntaxHighlighter, context: vscode.ExtensionContext, webviewPanel: vscode.WebviewPanel) => {
  const customHighlightJsAdapter = Opal.klass(Opal.nil, highlightjsBuiltInSyntaxHighlighter, 'CustomHighlightJsAdapter')
  // ...
}
```

Reply:
I almost understand :confounded: